File: Python/gate_six.py/parking_lot.py
def parking_car_lot():
    parking_space = []

    for car_space in range(1, 21):
        parking_space.append(car_space)

    return parking_space

def choose_parking_space(parking_space):

    for count in range(1, 21):
        if parking_space[count] == 0:
            parking_space[count] = 1
            return parking_space

    return parking_space


def park_at_slot(parking_space, slot_number):

    if slot_number < 1 or slot_number > 20:
        return "Invalid input."


    if(parking_space[slot_number - 1] == 0):
      parking_space[slot_number - 1] = 1
      return f"Car parked at slot {slot_number}"

    else:
        return "Slot already occupied."

def cars_parked_leaving_slot(slot_number):

    if (slot_number < 1 || slot_number > 20){
        return "Invalid input"
    
    if (parking_space[slot_number - 1] == 1):
        parking_space[slot_number - 1] = 0; 
        return f"Car removed from slot {slot_number}"

    else:
        return "Slot is already empty"
     

def display_parking_status(slot):
    result = ""

    for count in range(20):
        if slot[count] == 0:
            result += f"Slot {count + 1} is empty\n"
        else:
            result += f"Slot {count + 1} is filled\n"

    return result

File: Java/Gate Six/ParkingLot.java
public class ParkingLot{

    public static int [] parkingLotSlot(int [] parkingLotSpace){

        int [] parkingSpace = new int [20];

        for(int carSpace = 0; carSpace < 20; carSpace++){
            if(parkingSpace[carSpace] >= 0){
                parkingSpace[carSpace] = carSpace + 1;
                    return parkingSpace;
            }

            else if(parkingSpace[carSpace] < 0){
                return parkingSpace[carSpace] = -1;
             }
        }
        return parkingSpace;
    }


    public static int [] chooseYourParkingSpace(int [] parkingSpace){

        for(int count = 0; count < 20; count++){
                if (parkingSpace[count] == 0){
                    parkingSpace[count] = 1;
                    return parkingSpace;
                }
            }
        return parkingSpace;
        }

    
    public static String parkAtSlot(int [] parkingSpace, int slotNumber){

        if(slotNumber < 1 || slotNumber > 20){
            return "Invalid input.";
        }

        if(parkingSpace[slotNumber - 1] == 0){
          parkingSpace[slotNumber - 1] = 1;
            return "Car parked at slot" + slotNumber;
        }

        else{
            return "Slot already occupied.";
        }    
    }


    public static String carsParkedLeavingSlot(int [] slotNumber, int[] parkingSpace){
        if (slotNumber < 1 || slotNumber > 20){
            return "";
        }

        if (parkingSpace[slotNumber - 1] == 1){ 
            parkingSpace[slotNumber - 1] = 0; 
            return "Car removed from slot " + slotNumber;
        } 

        else{
            return "Slot is already empty";
        }
    }


    public static int [] displayParkingStatus(int [] slots){

        for(int count = 0; count < 20; count++){
            if(slot[count] == 0){
                return "The slot is" + count + slot + "is empty";
            } 
            else if(slot[count] == 1){
                return "The slot is" + count + slot + "is filled.";
            }
        }             
    }

}
